File: 40/answer.py
from functools import reduce
import time

start_time = time.time()

tempArray = []
nthArray = [1,10,100,1000,10000,100000,1000000]

concString = ''

for x in range(nthArray[-1]):
    concString += str(x + 1)

counter = 0

for y in concString:
    counter += 1
    if counter in nthArray:
        tempArray.append(int(y))

arrayMultiply = reduce(lambda x, y: x*y, tempArray)

print("Numbers from Nth Position:", tempArray)
print("Multiplied Product:",arrayMultiply)
print("Executed In:",(time.time() - start_time), 'seconds.') #STO 1557571

File: 40/answer.js
let counter = 0
let tempArray = []
let concString = ''
let nthArray = [1,10,100,1000,10000,100000,1000000]


for (let x = 0; x < nthArray[nthArray.length - 1]; x++) {
  concString += String(x + 1)
}

for (let y = 0 ; y < concString.length; y ++) {
  counter += 1
  let number = concString.charAt(y)
  if(nthArray.includes(counter)) {
    tempArray.push(parseInt(number))
  }
}

let total = tempArray.reduce((x, y) => x * y)

console.log(tempArray)
console.log(total)
